### DISCUSSION TITLE
obx kicks in ONLY after hot-reload!

### DISCUSSION THREAD

**Discussion by natgross** [jonataslaw/getx#1971] [Q&A] [answered]
In my controller I have a list defined: `final xEntitiesResultListMainDb = <TestEntity>[].obs;`
In a (db query) function I do:  `xEntitiesResultListMainDb.assignAll(newQueryResultList);`
In my stateful widget/view I inject the controller:    `final xController = Get.find<FirstGetxController>();`
Then a view wraps a List builder with obx, displaying a ListTile per item.
**The problem**:
Nothing happens when updating the obs list in the controller, even when using assignAll().
BUT! (This is the funny part.) If I hot reload (device connected to windows machine on vscode), voila! The (obx) list appears promptly (as expected in the first place)  _without_ any further input. 

The same phenomenon occurs with other variables contained in the entities in the list. They appear as expected ONLY after a hot reload.

Thanks,

**Comment by natgross** [comment, marked as answer]
AND the answer is...
You shouldn't build an obx wrapped widget as an argument to an else statement.
Here is a snippet of _incompatible_ code:
```
           Container(                
                child: isSomeConditionTrue?
                    Text("-------------")
                    : Obx(
                        () => ListView.builder(...
```
When I removed the if-else, and had the Obx directly as the child, all is ok!

**Reply by AlimFreight** [reply]
But what if I need 'if' :), And I find out that even if you have 'if' in a function which is updating the [list] behaviour the same :(
